### ISSUE TITLE
Running NoDevFee globally

### ISSUE THREAD

**Issue by mehtaman** [Demion/nodevfee#44]
Hi,

Is there any way to run NoDevFee globally on a system e.g. it's always watching for a miner software?
If I am profit switching, the miner software changes often.
Can NoDevFee detect this without the .exe / .dll being in the same folder as the miner?

**Comment by angelics** [comment]
from my undestanding. (im no expert in programming)
yes if connection to devfee pool is unencrypted
yes if nodevfee is able to include both eth authorize + mining authorize / mining submit
yes if nodevfee is able to inject anytype of wallet address length and wallet indicator (eg. btc address, equihash address)
i tried the above i mentioned but failed.

**Comment by Freekers** [comment]
No. As far as I know, every miner process needs to have it's own 'nodevfee' executable in the same folder. So if you're running 6 different miner software on the same machine, you need 6x the 'nodevfee' executable in the same folder as the miner software.

**Comment by mehtaman** [comment]
Thats pretty much what I have running at the moment...
On system startup a batch file runs to call other batch files e.g.
@echo off
setx gpu...
call C:\...\claymore_eth_ndf.bat
call C:\...\claymore_neo_ndf.bat
call C:\...\claymore_crp_ndf.bat

Then onces nodevfee is running on all batch files miner starts up.

**Comment by Demion** [comment]
> - To work with 3rd party miners create `nodevfeeInject.txt` with your miner file name inside; run `nodevfee.exe` without parameters; run your miner as usual (without `nodevfee.exe` before miner). *Note: `nodevfee.exe` should keep running; `nodevfee.exe` `nodevfee.dll` and all config files should be in same directory as your miner.*

You can try external injection method. You can have single instance of `nodevfee.exe` and `nodevfeeInject.txt` in separate folder, but you still will need to have `nodevfee.dll` and all other settings text files in each miner folder, so its practically similar solution.

_I will think about how to make "profile switch" better._

**Comment by LostSoulfly** [comment]
The easy solution is to use https://github.com/Demion/nodevfee/issues/34

This works regardless which miner is running, assuming you have input the correct addresses in the config.
